tools/esmgen.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const builtins = require('repl')._builtinLibs;
+const fs = require('fs');
+
+/* eslint-disable no-template-curly-in-string */
+
+const test = [
+  'const check = (T1, T2) => {',
+  '  if (!T1) throw new Error(`${T1} == true`);',
+  '  if (!T2) throw new Error(`${T2} == true`);',
+  '};',
+];
+
+const headers = {
+  assert: '/* eslint-disable no-restricted-properties */\n',
+};
+
+for (const builtin of builtins) {
+  const m = require(builtin);
+  const descriptors = Object.getOwnPropertyDescriptors(m);
+  const keys = [];
+  for (const [name, d] of Object.entries(descriptors)) {
+    if (/(^_)|(_$)/.test(name))
+      continue;
+    if (!d.enumerable)
+      continue;
+    if (!d.value)
+      continue;
+
+    keys.push(name);
+  }
+  genTest(builtin, keys);
+  const src = `${headers[builtin] || ''}import ${builtin} from '${builtin}';
+
+export default ${builtin};
+
+${keys.map((k) => `export const ${k} = ${builtin}.${k};`).join('\n')}
+`;
+  fs.writeFile(`./lib/${builtin}.mjs`, src, (err) => {
+    if (err)
+      console.error(builtin, err);
+    else
+      console.log('built', builtin);
+  });
+}
+
+function genTest(name, keys) {
+  const key = keys.filter((k) => k !== 'constants')[0];
+  test.push(`import ${name}, { ${key} } from '${name}';`);
+  test.push(`check(${name}, ${key});`);
+}
+
+console.log(test.join('\n'));
```

Comment:
Is the assumption that this file is run periodically to update the `lib` folder? Is there a way to make this more automated? Why would you think this level of automation is preferable to runtime enumeration?

Reply:
hopefully it never needs to be used again, but i still included it since it might be useful at some point.